lib/utils.pm

```diff
@@ -902,6 +903,13 @@ sub install_all_from_repo {
     assert_script_run($exec_str);
 }
 
+sub gather_pidstat_data {
+    my ($pidstat_file) = @_;
+
+    script_run('pkill pidstat');
+    upload_logs($pidstat_file);
+}
+
```

Comment:
can you instead move that function to a new baseclass? then you can also define the post_fail_hook in the baseclass and use it there.